packages/ag-charts-community/src/api/preset/sparkline.ts

```diff
@@ -235,15 +242,27 @@ function axisPreset(
     return { type: defaultType };
 }
 
-function gridLinePreset(opts: AgSparklineAxisOptions | undefined, defaultEnabled: boolean): AgAxisGridLineOptions {
+function gridLinePreset(
+    opts: AgSparklineAxisOptions | undefined,
+    defaultEnabled: boolean,
+    sparkOpts: AgSparklineOptions
+): AgAxisGridLineOptions {
     const gridLineOpts: AgAxisGridLineOptions = {};
+    const isBarVertical = sparkOpts.type === 'bar' && sparkOpts.direction !== 'horizontal';
```

Comment:
We have a check for `isVertical` (called `swapAxes`) below. Maybe we could pass that boolean in instead

Reply:
that other check isn't quite the same, it checks for !bar || vertical. In this case, this must only apply to vertical bar charts (we don't currently default to showing the gridline for line or area charts)

Reply:
Simplified